### DISCUSSION TITLE
About solana transaction instruct decode.

### DISCUSSION THREAD

**Discussion by devlancer412** [community/community#20847] [Programming Help] [answered]
How I can decode instruction data in solana transaction?
for example instruction likes following:
```
'transaction': {
    'message': {
      'accountKeys': [
        'Ap5pxfhTsW8bW4SvbezbrGdaSWRDmNSMycgCu11ba4i',
        '2Fgjpc7bp9jpiTRKSVSsiAcexw8Cawbz7GLJu8MamS9q',
        '2TgaaVoHgnSeEtXvWTx13zQeTf4hYWAMEiMQdcG6EwHi',
        '31cKs646dt1YkA3zPyxZ7rUAkxTBz279w4XEobFXcAKP',
        '4rm5QCgFPm4d37MCawNypngV4qPWv4D5tw57KE2qUcLE',
        'AoQJZaoTePWwgvHG4MTtUZ3N9Eqkxny25vpFmJYzSAvM',
        '98pjRuQjK3qA6gXts96PqZT4Ze5QmnCmt3QYjhbUSPue',
        'ComputeBudget111111111111111111111111111111',
        'EBDRoayCDDUvDgCimta45ajQeXbexv7aKqJubruqpyvu',
        'mv3ekLzLbnVPNxjSKvqBpU3ZeZXPQdEC3bp5MDEBG68'
      ],
      'header': {
        'numReadonlySignedAccounts': 0,
        'numReadonlyUnsignedAccounts': 4,
        'numRequiredSignatures': 1
      },
      'instructions': [
        {
          'accounts': [],
          'data': '1Bt5KeS2ztXq',
          'programIdIndex': 7
        }, {
          'accounts': [6, 8, 2, 3, 1, 4, 5],
          'data': 'HRDyPWG4FgAygKRH',
          'programIdIndex': 9
        }
      ],
      'recentBlockhash': 'ESABqYgqgbdKcFYTa9JYd41eHNFkredntfXa6g4MSH8j'
    },
    'signatures': ['2UYUmRNzACtuQJVnEmy1SJMcNPno6on9XBEuj9yWzdGy8D3icPH6mf8CVX5TwhcNp6TM9wSkt1GZvfU2mgiy7dms']
  }
```

**Comment by Wuffle1125** [comment, marked as answer]
Hi @devlancer412 

you can review this url:
https://stackoverflow.com/questions/69432084/how-to-decode-a-solana-transaction

**Comment by Kage0923** [comment]
Hi @everyone

you can review this url:
https://stackoverflow.com/questions/69432084/how-to-decode-a-solana-transaction

**Comment by xingarr** [comment]
Hi @devlancer412

you can review this url:
https://stackoverflow.com/questions/69432084/how-to-decode-a-solana-transaction

**Reply by haoweig** [reply]
@devlancer412 how did you get  the above transaction results? did you use the solana cli?

**Comment by SuperDev313** [comment]
To decode instruction data in a Solana transaction, you need to understand the structure of the transaction and how to interpret the data fields within the instructions. 

The process can be like this:

Parsing the Transaction
Decoding the Base64 Data
Understanding the Program-specific Data Structure

**Comment by 0xsatoshi99** [comment]
Here is another solution.
https://solana.stackexchange.com/questions/9380/how-to-decode-data-instructions-with-javascript
